File: Korp_Teste_MarcosMaciel.Server/Models/NotaFiscal.cs
namespace Korp_Teste_MarcosMaciel.Server.Models;

public class NotaFiscal
{
    public int Id { get; set; }
    public int Numero { get; set; }
    public string Status { get; set; } = "Aberta";
    public List<NotaFiscalItem> Itens { get; set; } = new();
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;
    public DateTime AtualizadoEm { get; set; } = DateTime.UtcNow;

    public void Validar()
    {
        if (Numero <= 0)
            throw new InvalidOperationException("Número da nota fiscal é obrigatório.");

        if (string.IsNullOrWhiteSpace(Status))
            throw new InvalidOperationException("Status da nota fiscal é obrigatório.");

        if (Itens == null || Itens.Count == 0)
            throw new InvalidOperationException("A nota fiscal deve conter pelo menos um item.");

        if (Itens.Any(x => x.Quantidade <= 0))
            throw new InvalidOperationException("Cada item da nota fiscal deve ter quantidade maior que zero.");
    }
}

File: korp_teste_marcosmaciel.client/src/app/models/nota-fiscal.ts
export interface NotaFiscalItem {
  id?: number;
  produtoId: number;
  quantidade: number;
  produto?: {
    id: number;
    codigo: string;
    descricao: string;
    saldo: number;
  };
}

export interface NotaFiscal {
  id?: number;
  numero?: number;
  status?: 'Aberta' | 'Fechada';
  itens: NotaFiscalItem[];
  criadoEm?: string;
  atualizadoEm?: string;
}

export interface CreateNotaFiscalRequest {
  itens: NotaFiscalItem[];
}
